packages/node_modules/@webex/plugin-meetings/src/participants/util.js

```diff
@@ -0,0 +1,181 @@
+  if (!(participant && participant.devices && participant.devices.length)) {
+    return NOT_IN_MEETING;
+  }
+  const device = participant && participant.devices ? participant.devices[0] : null;
+  if (device) {
+    if (device.state === JOINED && (!device.intent || device.intent.type === NONE)) {
+      return IN_MEETING;
+    }
+    if (device.state === IDLE && device.intent && (device.intent.type === WAIT || device.intent.type === OBSERVE)) {
+      return IN_LOBBY;
+    }
+    if (device.state === LEFT || (device.state === IDLE && (!device.intent || device.intent.type === NONE))) {
+      return NOT_IN_MEETING;
+    }
+  }
+  return NOT_IN_MEETING;
+};
+
+ParticipantUtil.handledAddToLobbyMeetingUnjoinedLists = (participants, participant) => {
+  switch (ParticipantUtil.getParticipantStatus(participant)) {
+    case IN_LOBBY:
+      return participants.inLobbyList.push(participant);
+    case IN_MEETING:
+      return participants.inMeetingList.push(participant);
+    case NOT_IN_MEETING:
+      return participants.notInMeetingList.push(participant);
+    default:
+      return participants.notInMeetingList.push(participant);
+  }
+};
+
+ParticipantUtil.handleAddToHostList = (participants, participant, hostIdentity, selfUserIdentity) => {
+  if (hostIdentity
+  && hostIdentity !== selfUserIdentity) {
+    if (ParticipantUtil.isUser(participant)
+    && participant.identity === hostIdentity) {
+      return participants.hostList.push(participant);
+    }
+    if (ParticipantUtil.isDevice(participant)
+    && participant.associatedUsers && participant.associatedUsers.some((user) => (user.identity === hostIdentity))
+    ) {
+      return participants.hostList.push(participant);
+    }
+  }
+  return null;
+};
+
+ParticipantUtil.handleAddToSelfList = (participants, participant, selfUserIdentity) => {
+  if (ParticipantUtil.isUser(participant)
+  && participant.identity === selfUserIdentity) {
+    return participants.selfList.push(participant);
+  }
+  if (ParticipantUtil.isDevice(participant)
+  && participant.associatedUsers && participant.associatedUsers.some((u) => (u.identity === selfUserIdentity))) {
+    return participants.selfList.push(participant);
+  }
+  return null;
+};
+
+ParticipantUtil.constructParticipants = (locus, participants, selfUserIdentity, hostIdentity) => {
+  if (!(locus && locus.participants && locus.participants.length && participants)) {
+    throw new Error('Could not construct meeting participant list. Some value is not present');
+  }
+  forEach(locus.participants, (locusParticipant) => {
+    if (locusParticipant.hideInRoster) {
+      return;
+    }
+    if (ParticipantUtil.handleAddToSelfList(participants, locusParticipant, selfUserIdentity)) {
+      return;
+    }
+    if (ParticipantUtil.handleAddToHostList(participants, locusParticipant, hostIdentity, selfUserIdentity)) {
+      return;
+    }
+    ParticipantUtil.handledAddToLobbyMeetingUnjoinedLists(participants, locusParticipant);
```

Comment:
consider optimizing the order of the function calls here, and add an else { put into not meeting }
put a TODO